File: Segundo/PDOO/CivitasJava/src/main/java/civitas/CasillaJuez.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author jmramirez
 */
public class CasillaJuez extends Casilla{
    
    private int carcel;
    
    CasillaJuez(int nCarcel, String n) {
        super(n);
        carcel = nCarcel;
    }
    
    void recibeJugador(int actual, ArrayList<Jugador> todos) {
        if (jugadorCorrecto(actual, todos)) {
            informe(actual, todos);
            todos.get(actual).encarcelar(carcel);
        }
    }
    
    public String toString() {
        String ret = "la casilla ";
        ret += this.nombre;
        ret += ", de tipo Juez.";
        return ret;
    }
    
}

File: Segundo/PDOO/CivitasRuby/Civitas/CasillaJuez.rb
# encoding:utf-8


module Civitas

    class CasillaJuez < Casilla

        def initialize(numCarcel, n)
            super(n)
            @nCarcel = numCarcel
        end

        def informe(actual, todos)
            Diario.instance.ocurre_evento("El jugador " + todos.at(actual).nombre + " ha caído en una casilla de tipo Juez.")
        end

        def recibe_jugador(actual, todos)
            if jugadorCorrecto(actual, todos)
                informe(actual, todos)
                todos.at(actual).encarcelar(@nCarcel)
            end
        end

        def toString
            return "La casilla actual es " + @nombre + " y es de tipo Juez.\n"
        end

    end

end